### PR TITLE
cmake: Cleanup layers/CMakeLists.txt

### PR DESCRIPTION
**Pull request by juan-lunarg**
- Remove pointless macro AddVkLayer that makes reading/debugging
much more difficult.
- Remove TARGET_NAMES since it also makes reading/debugging much
more difficult
- Move VkLayers_Utils to appropriate location

### CODE REVIEW

File: layers/CMakeLists.txt [hunk 1]
```diff
@@ -320,50 +320,36 @@ if(BUILD_LAYERS)
...
-        elseif(UNIX)
-            install(FILES ${CMAKE_CURRENT_BINARY_DIR}/staging-json/${TARGET_NAME}.json
-                    DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/vulkan/explicit_layer.d)
-        endif()
-    endforeach()
+    if(WIN32)
+        install(FILES $<TARGET_FILE_DIR:VkLayer_khronos_validation>/VkLayer_khronos_validation.json DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    elseif(UNIX)
+        install(FILES ${CMAKE_CURRENT_BINARY_DIR}/staging-json/VkLayer_khronos_validation.json DESTINATION ${CMAKE_INSTALL_DATAROOTDIR}/vulkan/explicit_layer.d)
+    endif()
```
Comment (juan-lunarg) [inline layers/CMakeLists.txt:354]: Currently the installation logic is slightly buggy on UNIX platforms. I have to manually run `VkLayer_khronos_validation-staging-json` and then I can install. I'll address this in another PR.

**Comment by Tony-LunarG** [comment]
I believe VulkanTools counts on VkLayer_utils for layer_factory, api_dump, and other VT layers.

**Comment by juan-lunarg** [comment]
> I believe VulkanTools counts on VkLayer_utils for layer_factory, api_dump, and other VT layers.

Good catch, I moved the install call to the appropriate location then.

File: layers/CMakeLists.txt [hunk 2]
```diff
@@ -69,40 +69,9 @@ if(BUILD_LAYER_SUPPORT_FILES)
         generated/vk_extension_helper.h
         generated/vk_typemap_helper.h)
     install(FILES ${LAYER_UTIL_FILES} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/vulkan)
+    install(TARGETS VkLayer_utils DESTINATION ${CMAKE_INSTALL_LIBDIR})
```
Comment (juan-lunarg) [inline layers/CMakeLists.txt:72]: Moving it here makes the intent more clear.

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM

**Pull request merged**